cloudsim-plus/src/main/java/org/cloudbus/cloudsim/power/PowerMeter.java

```diff
@@ -0,0 +1,133 @@
+package org.cloudbus.cloudsim.power;
+
+import org.cloudbus.cloudsim.core.CloudSimEntity;
+import org.cloudbus.cloudsim.core.CloudSimTags;
+import org.cloudbus.cloudsim.core.Simulation;
+import org.cloudbus.cloudsim.core.events.SimEvent;
+import org.cloudbus.cloudsim.power.models.PowerModel;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.LinkedList;
+import java.util.List;
+import java.util.concurrent.Callable;
+
+import static org.cloudbus.cloudsim.core.CloudSimTags.POWER_MEASUREMENT;
+
+/**
+ * Periodically measures the current power usage of one or more {@link PowerAware} entities,
+ * according to a defined {@link #getMeasurementInterval() interval}, storing the results.
+ *
+ * @see #getPowerMeasurements()
+ * @since CloudSim Plus 6.0.0
+ */
+public class PowerMeter extends CloudSimEntity {
+
+    private final Callable<List<? extends PowerAware<? extends PowerModel>>> powerAwareEntitiesFn;
+
+    private double measurementInterval;
+    private double startTime;
+
+    private final List<PowerMeasurement> powerMeasurements = new LinkedList<>();
+
+    /**
+     * Initializes a {@link PowerMeter} with a function that returns a list of {@link PowerAware} entities on every measurement.
+     *
+     * This is useful if the list of entities varies during a simulation run.
+     */
+    public PowerMeter(Simulation simulation, Callable<List<? extends PowerAware<? extends PowerModel>>> powerAwareEntitiesFn) {
```

Comment:
I used this a lot in LEAF, but I think it's a general-purpose feature.

It's a useful feature if you want to measure the overall power usage of a set of entities that can change during the simulation. For example, in LEAF I simulated taxis that frequently joined or left the simulation. So if `getTaxis()` returns all taxis that are on the map right now, something like `new PowerMeter(simulation, getTaxis())` gives me a `PowerMeter` which returns the overall power usage of all taxis currently on the map at each measurement.

It can be used for any kind of dynamic filtering. You could for example create a `PowerMeter` that only measures the power usage of hosts that currently execute a certain VM or that are active etc.

Shall I include an example in the doc?

Reply:
I see. So maybe, we could use a more intuitive interface from the `java.util.function` package.
I think the `Supplier` interface is better because it provides a more meaningful name.

Reply:
Makes sense!